code/modules/logging/categories/log_category_misc.dm

```diff
@@ -65,3 +65,7 @@
 	category = LOG_CATEGORY_QDEL
 	// We want this human readable so it's easy to see at a glance
 	entry_flags = ENTRY_USE_DATA_W_READABLE
+
+/datum/log_category/cave_generation
+	category = LOG_CATEGORY_CAVE_GENERATION
+	config_flag = /datum/config_entry/flag/log_cave_generation
```

Comment:
This is the core fix to your problem.

```suggestion
```

**Explanation**
Yup you don't need it. To understand why you need to first know the purpose of an config flag. As the name states this var loads a value from your config file and which you can then store in this log category.

The problem? You have not specified any value for this in your `/config/config.txt` file.

For example in your case your config flag is 
`config_flag = /datum/config_entry/flag/log_cave_generation`

so taking the last part of your flag as the config entry name i.e. `log_cave_generation` You need to have an entry as such in `/config/config.txt` file. like such

```
## this value is used for logging ore vent stuff
log_cave_generation = "some value"
```

But you never did this. Because of that when your config file attempted to load the value for your flag it could not find any so it was left with a null value which is why during validation of your config flag in this code here

https://github.com/tgstation/tgstation/blob/86b0aee29503760ddf5414b8abe819550190a902/code/modules/logging/log_holder.dm#L142-L144

It saw that your flag had no value attached to it so it "Disabled your entire logging category". Which is why whevever you tried to log with this category this check failed

https://github.com/tgstation/tgstation/blob/86b0aee29503760ddf5414b8abe819550190a902/code/modules/logging/log_holder.dm#L319-L320

It early returned & never bothered logging.

**Conclusion**
Use a config flag only if you plan to load a value from the `/config/config.txt` and use it in your code lime this

`var/my_value = glob.config.Get(/datum/config_entry/blah)`

If you have no need for such values(like in our case) don't bother using them